lib/datasource/orb/index.ts

```diff
@@ -1,64 +1,58 @@
 import { logger } from '../../logger';
-import * as packageCache from '../../util/cache/package';
-import { Http } from '../../util/http';
+import { cache } from '../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
 import type { GetReleasesConfig, ReleaseResult } from '../types';
 import type { OrbRelease } from './types';
 
-export const id = 'orb';
-export const defaultRegistryUrls = ['https://circleci.com/'];
-export const customRegistrySupport = false;
+export class OrbDatasource extends Datasource {
+  static readonly id = 'orb';
 
-const http = new Http(id);
-
-/**
- * orb.getReleases
- *
- * This function will fetch an orb from CircleCI and return all semver versions.
- */
-export async function getReleases({
-  lookupName,
-  registryUrl,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
-  logger.debug({ lookupName }, 'orb.getReleases()');
-  const cacheNamespace = 'orb';
-  const cacheKey = lookupName;
-  const cachedResult = await packageCache.get<ReleaseResult>(
-    cacheNamespace,
-    cacheKey
-  );
-  // istanbul ignore if
-  if (cachedResult) {
-    return cachedResult;
-  }
-  const url = `${registryUrl}graphql-unstable`;
-  const body = {
-    query: `{orb(name:"${lookupName}"){name, homeUrl, versions {version, createdAt}}}`,
-    variables: {},
-  };
-  const res: OrbRelease = (
-    await http.postJson<{ data: { orb: OrbRelease } }>(url, {
-      body,
-    })
-  ).body.data.orb;
-  if (!res) {
-    logger.debug({ lookupName }, 'Failed to look up orb');
-    return null;
+  constructor() {
+    super(OrbDatasource.id);
   }
-  // Simplify response before caching and returning
-  const dep: ReleaseResult = {
-    releases: null,
-  };
-  if (res.homeUrl?.length) {
-    dep.homepage = res.homeUrl;
+
+  customRegistrySupport = false;
+
+  defaultRegistryUrls = ['https://circleci.com/'];
+
+  @cache({
+    namespace: `datasource-${OrbDatasource.id}`,
+    key: ({ lookupName }: GetReleasesConfig) => lookupName,
+    ttlMinutes: 15,
```

Comment:
@rarkins Two questions:

I added `datasource-` as a prefix for the namespace to bring it in line with the other datasource caches. Is this okay, or would you prefer not to have the cache miss?

Can I raise the `ttlMinutes` to the default of 30, or shall I keep it as 15?

Reply:
i think we should use default ttl. good idea to unify cache namespaces for datasources

Reply:
Okay I'm keeping the `datasource-orb` namespace, and set the ttl to use the default value (30 minutes)